src/openms/source/CHEMISTRY/TheoreticalSpectrumGenerator.cpp

```diff
@@ -778,8 +799,7 @@ namespace OpenMS
       p.setIntensity(pre_int_H2O_);
       if (add_metainfo_)
       {
-        String ion_name("[M+H]-H2O" + String((Size)abs(charge), '+'));
-        ion_names.push_back(ion_name);
+        ion_names.push_back(ion_name_h2o);
         charges.push_back(charge);
       }
       spectrum.push_back(p);
```

Comment:
there are still a few places where push_back for Peak1D is used. Replace with emplace_back(mz,int) ?

Reply:
Replaced all occurences